packages/payload/src/admin/forms/Field.ts

```diff
@@ -1,24 +1,28 @@
+import type { MarkOptional } from 'ts-essentials'
+
 import type { User } from '../../auth/types.js'
 import type { Locale } from '../../config/types.js'
-import type { Validate } from '../../fields/config/types.js'
+import type { ClientField, Validate } from '../../fields/config/types.js'
 import type { DocumentPreferences } from '../../preferences/types.js'
-import type { ErrorProps } from './Error.js'
-import type { FieldDescriptionProps } from './FieldDescription.js'
-import type { LabelProps } from './Label.js'
+import type { FieldDescriptionClientProps } from './Description.js'
+import type { FieldErrorClientProps } from './Error.js'
+import type { FieldLabelClientProps } from './Label.js'
 
-// TODO: Check if we still need this. Shouldnt most of it be present in the field type?
-export type FormFieldBase = {
-  readonly descriptionProps?: FieldDescriptionProps
+export type FormFieldBase<
+  T extends MarkOptional<ClientField, 'type'> = MarkOptional<ClientField, 'type'>,
```

Comment:
What do you think about generally naming all generics, so `TField` instead of `T`? Gives me nicer suggestions when I hover over the type, as it immediately tells me what it does

Reply:
Yep, I like it